content/intro-to-storybook/ember/en/simple-component.md

```diff
@@ -0,0 +1,223 @@
+
+```javascript
+// .storybook/main.js
+
+module.exports = {
+  stories: ['../app/components/**/*.stories.js'],
+  addons: ['@storybook/addon-actions', '@storybook/addon-links'],
+};
+```
+
+Once we’ve done this, restarting the Storybook server should yield test cases for the three Task states:
+
+<video autoPlay muted playsInline loop>
+  <source
+    src="/intro-to-storybook//inprogress-task-states.mp4"
+    type="video/mp4"
+  />
+</video>
+
+## Build out the states
+
+Now we have Storybook setup, styles imported, and test cases built out, we can quickly start the work of implementing the HTML of the component to match the design.
+
+The component is still basic at the moment. First write the code that achieves the design without going into too much detail:
+
+```handlebars
+
+{{!--  app/components/Task.hbs --}}
+<div class="list-item {{@task.state}}">
+  <label class="checkbox">
+    <input type="checkbox" disabled name="checked" checked={{this.isTaskArchived}}/>
+    <span class="checkbox-custom" onclick={{this.handleArchiveTask}}/>
+  </label>
+  <div class="title">
+    <input type="text" readonly value={{@task.title}} placeholder="Input title" />
+  </div>
+  <div class="actions">
+    {{#if (not-eq @task.state "TASK_ARCHIVED")}}
+      <a href="/" onclick={{this.handlePinTask}}><span class="icon-star"  /></a>
+    {{/if}}
+  </div>
+</div>
+```
+
+Then we'll need create a new file called `app/components/Task.js` with the following:
+
+```js
+// app/components/Task.js
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+export default class Task extends Component {
+  constructor() {
+    super(...arguments);
+  }
+  // computed property for the component (to assign a value to the task state checkbox)
+  get isTaskArchived() {
+    return this.args.task.state === 'TASK_ARCHIVED';
+  }
+  // actions to emulate the click handlers to call the actions passed down to the component
+  @action
+  handlePinTask(e) {
+    e.stopPropagation();
+    e.preventDefault();
+    if (this.args.pinTask) {
+      this.args.pinTask(this.args.task.id);
+    }
+  }
+  @action
+  handleArchiveTask() {
+    if (this.args.archiveTask) {
+      this.args.archiveTask(this.args.task.id);
+    }
+  }
+}
+```
+
+The additional markup from above combined with the CSS we imported earlier yields the following UI:
+
+<video autoPlay muted playsInline loop>
+  <source
+    src="/intro-to-storybook/finished-task-states.mp4"
+    type="video/mp4"
+  />
+</video>
+
+## Component built!
+
+We’ve now successfully built out a component without needing a server or running the entire frontend application. The next step is to build out the remaining Taskbox components one by one in a similar fashion.
+
+As you can see, getting started building components in isolation is easy and fast. We can expect to produce a higher-quality UI with fewer bugs and more polish because it’s possible to dig in and test every possible state.
```

Comment:
Does storyshots not work with Ember at the moment @jonniebigodes?

Reply:
@tmeasday no it doesn't i checked and according to [this](https://github.com/storybookjs/storybook/blob/master/ADDONS_SUPPORT.md) it doesn't support the addon.

Reply:
Ok, thanks, sorry I could have checked that myself 😊